include/ginkgo/core/matrix/csr.hpp

```diff
@@ -848,6 +848,30 @@ class Csr : public EnableLinOp<Csr<ValueType, IndexType>>,
         this->make_srow();
     }
 
+    /**
+     * Scales the matrix with a scalar.
+     *
+     * @param alpha  The entire matrix is scaled by alpha. alpha has to be a 1x1
+     * Dense matrix.
+     */
+    void scale(const LinOp *alpha)
+    {
+        auto exec = this->get_executor();
```

Comment:
I think you can pull the 1x1 assertions out here as well, since this should hold also for all potential subclasses (even though deriving from Csr might not be the best idea, but still).

Reply:
Does it make sense to have a `Scalable` mix-in for this?